### PR TITLE
Hit the api

### PR DESCRIPTION
**Pull request by rebaz36**
- Initialize API
- GET data from the API
- POST data to the API

### CODE REVIEW

File: src/modules/mainClass.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      mode: 'cors',
+      headers: { 'Content-type': 'application/json; charset=UTF-8' },
+      body: JSON.stringify({ name: this.gameName }),
+    };
+    fetch(this.url, option)
+      .then((response) => response.json())
+      .then((data) => {
+        this.currentGame = data.result.substring(13, 34);
+      })
+      .catch(() => new Error('Error in newPost'));
```
Comment (Whoistolu) [inline src/modules/mainClass.js:21]: - [ ] [REQUIRED] Great implementation @rebaz36, however, you need to use the `async` and `await` with the fetch method to send and receive data as stated by the [project requirement](https://github.com/microverseinc/curriculum-javascript/blob/main/leaderboard/m2_send_receive_data.md).

**Review by Whoistolu — CHANGES_REQUESTED** [review]
Hi @rebaz36 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

- [ ]  _Check the inline comments under the review for sections to improve._

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

### N/A

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/mainClass.js (hunk 1 above)
Comment (rebaz36) [reply src/modules/mainClass.js]: Thanks @Whoistolu, i seem to have missed this requirement when reading the project details.
appreciate you bringing this to my attention

Comment (rloterh) [reply src/modules/mainClass.js]: This is well resolved! ✔️

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Rebaz,  

Your project is complete! There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉🌟 

## Highlights 👍
- Promise feature properly implemented

<br>
Keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**